torchtune/rlhf/loss/dpo.py

```diff
@@ -231,3 +231,91 @@ def forward(
         rejected_rewards = self.beta * (policy_rejected_logps).detach()
 
         return losses, chosen_rewards, rejected_rewards
+
+
+class KTOLoss(nn.Module):
+    """
+    KTO: Kahneman-Tversky Optimization: https://arxiv.org/abs/2402.01306
+    Intuition from the paper:
+
+        The effectiveness of SimPO is attributed to a key design: using the average log probability of a sequence as
+        the implicit reward. Additionally, we introduce a target reward margin to the Bradley-Terry objective to
+        encourage a larger margin between the winning and losing responses, further enhancing the algorithm's performance.
+
```

Comment:
It looks like this has been copied from the SimPO loss?

Reply:
Oh, I used SimPO docs as template and forgot to add my docs in this part. Other comments are about KTO

Reply:
Fixed